File: java_projects/glue_code/commons-graph/src/main/java/org/apache/commons/graph/VertexPair.java
package org.apache.commons.graph;

import java.io.Serializable;
import org.graalvm.polyglot.Value;

public final class VertexPair<V> implements Serializable {
  private static final long serialVersionUID = 2333503391707156055L;
  private static Value clz = ContextInitializer.getPythonClass("/VertexPair.py", "VertexPair");
  private Value obj;

  public VertexPair(Value obj) {
    this.obj = obj;
  }

  public Value getPythonObject() {
    return obj;
  }

  public String toString() {
    //
    // return format("[head=%s, tail=%s]", head, tail);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("toString").as(String.class);
  }

  public int hashCode() {
    //
    // final int prime = 31;
    // return hash(1, prime, head, tail);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("hashCode").as(int.class);
  }

  public boolean equals(Object obj) {
    //
    // if (this == obj) {
    // return true;
    // }
    //
    // if (obj == null || getClass() != obj.getClass()) {
    // return false;
    // }
    //
    // @SuppressWarnings("unchecked") // equals() invoked against only same VertexPair type
    // VertexPair<V> other = (VertexPair<V>) obj;
    // return eq(head, other.getHead()) && eq(tail, other.getTail());
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("equals", obj).as(boolean.class);
  }

  public V getTail() {
    //
    // return tail;
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("getTail").as(V.class);
  }

  public V getHead() {
    //
    // return head;
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("getHead").as(V.class);
  }

  public VertexPair(V head, V tail) {
    //
    // this.head = checkNotNull(head, "Impossible to construct a Vertex with a null head");
    // this.tail = checkNotNull(tail, "Impossible to construct a Vertex with a null tail");
    //

    this.obj = clz.invokeMember("__init__", head, tail);
  }
}

File: data/skeletons/commons-graph/src/main/org/apache/commons/graph/VertexPair.py
# Imports Begin
from src.main.org.apache.commons.graph.utils.Objects import *
from src.main.org.apache.commons.graph.utils.Assertions import *
import typing

# Imports End


class VertexPair(Serializable, VertexPair):

    # Class Fields Begin
    __serialVersionUID: int = None
    __head: typing.Any = None
    __tail: typing.Any = None
    # Class Fields End

    # Class Methods Begin
    def toString(self) -> str:
        pass

    def hashCode(self) -> int:
        pass

    def equals(self, obj: typing.Any) -> bool:
        pass

    def getTail(self) -> typing.Any:
        pass

    def getHead(self) -> typing.Any:
        pass

    def __init__(self, head: typing.Any, tail: typing.Any) -> None:
        pass

    # Class Methods End
